packages/webapp/src/components/Forms/GeneralRevenue/index.jsx

```diff
@@ -12,93 +12,151 @@
+//import useHookFormPersist from '../../../containers/hooks/useHookFormPersist';
 
 const GeneralRevenue = ({
   onSubmit,
   title,
-  dateLabel,
-  customerLabel,
   currency,
   sale,
   useHookFormPersist,
   persistedFormData,
-  onClickDelete,
+  view,
+  handleGoBack,
+  onClick,
+  revenueTypeOptions,
+  onTypeChange,
+  buttonText,
+  customFormChildrenDefaultValues,
+  useCustomFormChildren = () => {},
+  revenueType,
+  revenueTypes,
+  onRetire,
 }) => {
   const { t } = useTranslation();
+  const [isDeleting, setIsDeleting] = useState(false);
 
-  const {
-    register,
-    handleSubmit,
-    formState: { errors, isValid },
-    getValues,
-  } = useForm({
+  const reactHookFormFunctions = useForm({
     mode: 'onChange',
     defaultValues: {
       [SALE_DATE]:
         (sale?.[SALE_DATE] && getLocalDateInYYYYDDMM(new Date(sale[SALE_DATE]))) ||
         persistedFormData?.[SALE_DATE] ||
         getLocalDateInYYYYDDMM(),
       [SALE_CUSTOMER]: sale?.[SALE_CUSTOMER] || persistedFormData?.[SALE_CUSTOMER] || '',
-      [VALUE]: !isNaN(sale?.[VALUE]) ? sale[VALUE] : null,
-      [NOTE]: sale?.[NOTE] || '',
+      [REVENUE_TYPE_ID]:
+        revenueTypeOptions?.find((option) => option.value === sale?.revenue_type_id) ||
+        revenueTypeOptions?.find(
+          (option) => option.value === persistedFormData?.[REVENUE_TYPE_ID],
+        ) ||
+        null,
+      [VALUE]: !isNaN(sale?.[VALUE])
+        ? sale[VALUE]
+        : !isNaN(persistedFormData?.[VALUE])
+        ? persistedFormData[VALUE]
+        : null,
+      [NOTE]: sale?.[NOTE] || persistedFormData?.[NOTE] || '',
+      ...customFormChildrenDefaultValues,
     },
   });
 
-  useHookFormPersist(getValues);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    getValues,
+    watch,
+    control,
+    reset,
+    setValue,
+  } = reactHookFormFunctions;
+
+  //useHookFormPersist(getValues);
+  const selectedTypeOption = watch(REVENUE_TYPE_ID);
+
+  const readonly = view === 'read-only';
+  const disabledInput = readonly;
+  const disabledButton = !isValid && !readonly;
+
+  // TODO: LF-3680 fix isDirty for dynamic fields
+  // const disabledButton = (!isValid || !isDirty) && !readonly;
+
+  const customChildren = useCustomFormChildren(
+    reactHookFormFunctions,
+    sale,
+    currency,
+    disabledInput,
+    revenueTypes,
+    selectedTypeOption,
+  );
+
+  // Separating these into separate vs prop rendered nodes prevents form submission onClick for noSubmitButton
+  const noSubmitButton = (
+    <Button
+      color={'secondary'}
+      fullLength
+      disabled={disabledButton}
+      onClick={onClick}
+      type={'button'}
+    >
+      {' '}
+      {buttonText}
+    </Button>
+  );
+  const submitButton = (
+    <Button color={'primary'} fullLength disabled={disabledButton} type={'submit'}>
+      {' '}
+      {buttonText}
+    </Button>
+  );
```

Comment:
Here I'd suggest that we could consolidate all these props (buttonText, onSubmit, onClick) on a single `buttons` prop. This could either be an object specifying what each button should be like:

```js
buttonsSpec={[ 
    {
        type: {'submit'}, color: {'primary'}, text: 'Some text' },
    },
    {
        color: {'secondary'}, text: 'Some other text' },
    },
]}
```

or maybe an array of components

```js
buttonsSpec={[ 
    (props) => <Button color={'primary'} type={'submit'} {...props}>Some text<Button>,
    (props) => <Button color={'secondary'} {...props}>Some other text<Button>,
]}
```

and then in the component

```js
buttonGroup={
  <>
    {buttons.map((ButtonComponent) => (
      <ButtonComponent disabled={disabledInput} fullLength />
    ))}
  </>
}
```

The advantage with this approach is that if we give parent components the flexibility of setting as many buttons as they need and with the properties that they need, so it's a bit more scalable if at any point we need more buttons. It also allows for different button texts if any component needs more than one button without adding extra props, and consolidates all of the button related props in one which is a bit easier to read IMO.

Alternatively, if we keep the current approach, I'd suggest renaming `noSubmitButton` to something like `secondaryButton`, and giving `onClick` a name that's a bit more specific too.

Reply:
So I did try programmatically making/building the button with conditionals using `view` but the submit `onClick` would fire even if it shouldn't have rendered as such.

I think we really only ever want 1 button visible at a time. It seems like we have been moving away from a multi-button approach. So I I think secondary would make sense if it was a second visible button. But since it is really to signify the button shouldn't submit I am not sure.

Reply:
Come to think of it, why do we want the button not to submit? If we're using this for the edit form, wouldn't we want the Edit action to be the submit action in that case?

Reply:
For the edit action yes, for the readonly action wouldn't it be weird to submit the form just to unlock the editable fields? 

I think the core issue for me could be UI/UX related --- should the form submit button be the same button that unlocks the fields? Or should it be visually different like a pen icon on the input or a toggle in the top right or something.